app/assets/javascripts/discourse/app/components/topic-map/private-message-map.gjs

```diff
@@ -0,0 +1,215 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import avatar from "discourse/helpers/bound-avatar-template";
+import { groupPath } from "discourse/lib/url";
+import dIcon from "discourse-common/helpers/d-icon";
+import I18n from "discourse-i18n";
+import and from "truth-helpers/helpers/and";
+
+export default class PrivateMessageMap extends Component {
+  @service site;
+  @tracked isEditing = false;
+
+  get participantsClasses() {
+    return !this.isEditing &&
+      this.site.mobileView &&
+      this.args.postAttrs.allowedGroups.length > 4
+      ? "participants hide-names"
+      : "participants";
+  }
+
+  get canInvite() {
+    return this.args.postAttrs.canInvite;
+  }
+
+  get canRemove() {
+    return (
+      this.args.postAttrs.canRemoveAllowedUsers ||
+      this.args.postAttrs.canRemoveSelfId
+    );
+  }
+
+  get canShowControls() {
+    return this.canInvite || this.canRemove;
+  }
+
+  get actionAllowed() {
+    return this.canRemove ? this.toggleEditing : this.args.showInvite;
+  }
+
+  get actionAllowedLabel() {
+    if (this.canInvite && this.canRemove) {
+      return "private_message_info.edit";
+    }
+    if (!this.canInvite && this.canRemove) {
+      return "private_message_info.remove";
+    }
+    return "private_message_info.add";
+  }
+
+  @action
+  toggleEditing() {
+    this.isEditing = !this.isEditing;
+  }
+
+  <template>
+    <div class={{this.participantsClasses}}>
+      {{#each @postAttrs.allowedGroups as |group|}}
+        <PmMapUserGroup
+          @model={{group}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @removeAllowedGroup={{@removeAllowedGroup}}
+        />
+      {{/each}}
+      {{#each @postAttrs.allowedUsers as |user|}}
+        <PmMapUser
+          @model={{user}}
+          @isEditing={{this.isEditing}}
+          @canRemoveAllowedUsers={{@postAttrs.canRemoveAllowedUsers}}
+          @canRemoveSelfId={{@postAttrs.canRemoveSelfId}}
+          @removeAllowedUser={{@removeAllowedUser}}
+        />
+      {{/each}}
+    </div>
+
+    {{#if this.canShowControls}}
+      <div class="controls">
+        <DButton
+          @action={{this.actionAllowed}}
+          @label={{this.actionAllowedLabel}}
+          class="btn btn-default add-remove-participant-btn"
+        />
+
+        {{#if (and this.canInvite this.isEditing)}}
+          <DButton
+            @action={{@showInvite}}
+            @icon="plus"
+            class="btn btn-default no-text btn-icon add-participant-btn"
```

Comment:
I think some of these classes come by default in `<DButton>`, eg btn, btn-default, no-text, btn-icon, might be worth removing to double check.